processing/src/main/java/org/apache/druid/frame/processor/SuperSorter.java

```diff
@@ -811,15 +797,16 @@ private void cleanUp()
 
     outputsReadyByLevel.clear();
     inputBuffer.clear();
-
-    for (FrameFile frameFile : penultimateFrameFileCache.values()) {
-      CloseableUtils.closeAndSuppressExceptions(
-          frameFile,
-          e -> log.warn(e, "Could not close intermediate file [%s]", frameFile.file())
-      );
+    for (Map.Entry<String, PartitionedOutputChannel> cleanupEntry :
+        levelAndRankToReadableChannelMap.entrySet()) {
+      try {
+        cleanupEntry.getValue().getReadableChannelSupplier().get().close();
+      }
+      catch (IOException e) {
```

Comment:
On [Line 642](https://github.com/apache/druid/pull/13368/files#diff-6ab806ec1a5129c4c908965096a09b749ed597101e5f74b84fb82675bda8f8b2R642) we are warning but here we are throwing if close() throws an error. Why is that so?

Reply:
thanks for pointing that - throwing in this case too now